drivers/lvutil.py

```diff
@@ -138,6 +138,34 @@ def __exit__(self, exc_type, value, traceback):
             self.lock.release()
 
 
+LVM_RETRY_ERRORS = [
+    "Incorrect checksum in metadata area header"
+]
+
+
+def lvmretry(func):
+    def check_exception(exception):
+        retry = False
+        for error in LVM_RETRY_ERRORS:
+            if error in exception.reason:
+                retry = True
+        return retry
+
+    def decorated(*args, **kwargs):
+        retry = True
+        while retry:
```

Comment:
This will result in an infinite loop if the metadata is truly corrupted. Probably needs a counter and give up after N retries (otherwise we'll just fill /var/log whenever we got corruption, potentially even loosing the original reason due to aggressively triggering log rotation).

Reply:
(marked as 'changes required' to fix the infinite loop)